lib/include/cne_mutex_helper.h

```diff
@@ -30,19 +30,35 @@ extern "C" {
 static inline int
 cne_mutex_create(pthread_mutex_t *mutex, int flags)
 {
-    int ret = EFAULT;
-
-    if (mutex) {
-        pthread_mutexattr_t attr;
-
-        ret = pthread_mutexattr_init(&attr);
-        if (ret == 0) {
-            ret = pthread_mutexattr_settype(&attr, flags);
-            if (ret == 0) {
-                ret = pthread_mutex_init(mutex, &attr);
-                if (ret == 0 && pthread_mutexattr_destroy(&attr) == 0)
-                    return 0;
-            }
+    pthread_mutexattr_t attr;
+    int inited = 0, ret = EFAULT;
+
+    if (!mutex)
+        goto err;
+
+#define __do(_exp)    \
+    do {              \
+        ret = _exp;   \
+        if (ret)      \
+            goto err; \
+    } while (0 /* CONSTCOND */)
+
+    __do(pthread_mutexattr_init(&attr));
```

Comment:
I don't like the custom macro definition just to check a few failure cases, but if you strongly prefer this way I'll defer to you.

Reply:
In my head it seems cleanup and simpler. I know what the macro does and it is done the same way everywhere. I am going to keep it.